sensors/dht.py

```diff
@@ -0,0 +1,78 @@
+import RPi.GPIO as GPIO
+import typing
+import time
+import enum
+import random
+
+
+DHTLIB_DHT11_WAKEUP = 0.020
+DHTLIB_TIMEOUT = 0.0001
+
+
+class DHTCode(enum.Enum):
+    DHTLIB_OK = enum.auto()
+    DHTLIB_ERROR_CHECKSUM = enum.auto()
+    DHTLIB_ERROR_TIMEOUT = enum.auto()
+
+
+class DHTReading(typing.NamedTuple):
+    humidity: int
+    temperature: float
+    code: DHTCode
+
+
+def read_dht(pin: int, wakeup_delay = DHTLIB_DHT11_WAKEUP, timeout = DHTLIB_TIMEOUT):
+        mask = 0x80
+        idx = 0
+        bits = [0,0,0,0,0]
+        GPIO.setup(pin,GPIO.OUT)
+        GPIO.output(pin,GPIO.LOW)
+        time.sleep(wakeup_delay)  # FIXME: Can this be removed/replaced with async sleep?
```

Comment:
`wakeup_delay` is by default `DHTLIB_DHT11_WAKEUP` which is set to `0.020`. That's a small enough time period that making it async doesn't matter.